### DISCUSSION TITLE
Custom spi is not displaying in the keycloak UI - keycloak version is 23.0.0

### DISCUSSION THREAD

**Discussion by navyademo1** [keycloak/keycloak#28402] [Q&A]
Hi,
I am trying to create a custom spi for local db integration into keycloak, but after deployment in the /providers folder for some reason I couldn't able to see the created spi in my keycloak UI. Can any one help me with this.

UserRegistrationProvider:

import org.keycloak.events.Event;
import org.keycloak.events.EventListenerProvider;
import org.keycloak.events.admin.AdminEvent;
import org.keycloak.models.KeycloakSession;
import org.slf4j.LoggerFactory;

public class UserRegistrationProvider implements EventListenerProvider{

private KeycloakSession session;

private static final org.slf4j.Logger log =LoggerFactory.getLogger(UserRegistrationProvider.class);

public UserRegistrationProvider(KeycloakSession session) {
	this.session=session;
}

@Override
public void close() {
	
}

@Override
public void onEvent(Event event) {
	
}

@Override
public void onEvent(AdminEvent event, boolean includeRepresentation) {

	if("USER".equals(event.getResourceType().name()) && "CREATE".equals(event.getOperationType().name())) {
		log.info("user details:" + event.getRepresentation());
		
	}
}
}

UserRegistrationProviderFactory:

import org.keycloak.Config.Scope;
import org.keycloak.events.EventListenerProvider;
import org.keycloak.events.EventListenerProviderFactory;
import org.keycloak.models.KeycloakSession;
import org.keycloak.models.KeycloakSessionFactory;

public class UserRegistrationProviderFactory implements EventListenerProviderFactory{

private static final String PROVIDER_ID ="custom-user-registration";

@Override
public EventListenerProvider create(KeycloakSession session) {
	return new UserRegistrationProvider(session);
}

@Override
public void init(Scope config) {
	
}

@Override
public void postInit(KeycloakSessionFactory factory) {
	
}

@Override
public void close() {
	
}

@Override
public String getId() {
	return PROVIDER_ID;
}
}

I have also added the META-INF file in resources folder.

**Comment by codespearhead** [comment]
What do you mean by `local db integration` exactly?

Keycloak uses Hibernate, which is a JPA implementation, so [you can use any JPA-compliant driver enable it to connect to the database of your choice](https://www.keycloak.org/server/db).
